test/integration/scheduler/filters/filters_test.go

```diff
@@ -2440,3 +2440,202 @@ func TestPodAffinityMatchLabelKeyEnablement(t *testing.T) {
 	}
 
 }
+
+func TestNodeAffinityFilter(t *testing.T) {
+	pause := imageutils.GetPauseImageName()
+	tests := []struct {
+		name        string
+		nodes       []*v1.Node
+		incomingPod *v1.Pod
+		fit         bool
+	}{
+		{
+			name: "Pod fits when node matches label using In operator",
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-ssd").Label("disktype", "ssd").Obj(),
+			},
+			incomingPod: st.MakePod().Name("affinity-fitting-pod").
+				NodeAffinity(&v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{
+									{
+										Key:      "disktype",
+										Operator: v1.NodeSelectorOpIn,
+										Values:   []string{"ssd"},
+									},
+								},
+							},
+						},
+					},
+				}).Container(pause).
+				Obj(),
+			fit: true,
+		},
+		{
+			name: "Pod not fit when node not match label using In operator",
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-ssd").Label("disktype", "ssd").Obj(),
+			},
+			incomingPod: st.MakePod().Name("affinity-non-fitting-pod").
+				NodeAffinity(&v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{
```

Comment:
Do we need to test a case that pod has multiple keys and values but only 1 matched?